zebra-rpc/src/methods/get_block_template_rpcs/get_block_template.rs

```diff
@@ -305,7 +305,15 @@ pub fn generate_coinbase_and_roots(
     // Calculate block default roots
     //
     // TODO: move expensive root, hash, and tree cryptography to a rayon thread?
-    let default_roots = calculate_default_root_hashes(&coinbase_txn, mempool_txs, history_tree);
+    let chain_history_root = history_tree
+        .hash()
+        .or_else(|| {
+            (NetworkUpgrade::Heartwood.activation_height(network) == Some(height))
```

Comment:
Chain history roots were activated at Heartwood, it may be useful to add a comment here.

The null root hash is a reserved value for the chain history activation.

Reply:
I looked at this again. If we got no history tree for the Heartwood activation height, shouldn't we consider that an error and abort instead of using the null hash? (Even though we know that the null hash corresponds to the Heartwood activation height.)

Reply:
I think this part is confusing because the `height` parameter should be `next_height`

Reply:
Renamed the parameter in https://github.com/ZcashFoundation/zebra/commit/2ad072a19f1713090bbbaccabb5d3a7d381a55f1.

Reply:
> I think this part is confusing because the `height` parameter should be `next_height`, the condition in the `or_else()` would be true if we got no history tree for the Genesis activation height.

Thanks for the clarification. I didn't realize this.